idgen/idgen_test.go

```diff
@@ -0,0 +1,27 @@
+package idgen
+
+import (
+	"testing"
+)
+
+func TestBalancedGeneration(t *testing.T) {
+	const N = 10000
```

Comment:
In your experiments, what was the minimum N for the assert to always yield true? 10? 100?

Reply:
I assume there's no minimum, and that even this value is not guaranteed to always yield true, just be _very_ unlikely to not.

Reply:
> In your experiments, what was the minimum N for the assert to always yield true?

I didn't try numbers lower than 1000, but they probably work. The probability that balanced insertions produce smaller depth than unbalanced insertions increases exponentially with the number of nodes, so a large enough number like 10000 is good enough to make the test never fail (even though it is probabilistic). (Alternatively, the test can be seeded with a fixed seed so that the test is always reproducible.) 

If you are interested in ensuring balance, with high probability, even for very small numbers (like 20 nodes) then one can boost the probability by increasing the number of choices from 2 to 3 or 4. I don't think this is worth worrying about though, because when you have less than 20 nodes (which is the K bucket size), the distribution doesn't matter. Everyone gets everything.

Reply:
If you are concerned about getting 100 nodes very balanced, I think the current 2-choice algorithm is fine. But we can always make it a 4-choice, for peace of mind :)